complex/ft_msg.c

```diff
@@ -542,30 +556,80 @@ int ft_post_recv_bufs(void)
 	return 0;
 }
 
-int ft_recv_msg(void)
+int ft_recv_n_msg(int n)
 {
-	int credits, ret;
+	int credits, ret, recved = 0;
 
+post:
 	if (ft_rx_ctrl.credits > (ft_rx_ctrl.max_credits >> 1)) {
 		ret = ft_post_recv_bufs();
 		if (ret)
 			return ret;
 	}
 
-	credits = ft_rx_ctrl.credits;
-	do {
+	while (recved < n) {
+		credits = ft_rx_ctrl.credits;
 		ret = ft_comp_rx(FT_COMP_TO);
 		if (ret)
 			return ret;
+
+		recved += (ft_rx_ctrl.credits - credits);
```

Comment:
The function is intended to wait on at least n completions for easy use with bandwidth testing. This was true with the previous ft_recv_msg call as well but it shouldn't be a problem with how the tests are set up. Any call to ft_recv_n_msg is called with exactly how many messages it should be getting so there really should never be an 11th message if you are waiting on 10.